src/ethereum/shanghai/fork.py

```diff
@@ -751,7 +762,9 @@ def calculate_intrinsic_cost(tx: Transaction) -> Uint:
             access_list_cost += TX_ACCESS_LIST_ADDRESS_COST
             access_list_cost += len(keys) * TX_ACCESS_LIST_STORAGE_KEY_COST
 
-    return Uint(TX_BASE_COST + data_cost + create_cost + access_list_cost)
+    return Uint(TX_BASE_COST +
+                data_cost + 
+                create_cost + access_list_cost), tokens_in_calldata
```

Comment:
You'll need to update the return type of the function to `Tuple[Uint, Uint]` I think.